### PR TITLE
Check for file existence and check early

### LINKED ISSUE #2364: Convert uses of `stop()` to use `cli::cli_abort()`

### PR DESCRIPTION
**Pull request by jennybc**
Closes #2364 

I recently tried to build a non-existent file and would have preferred to learn this *before* installing the package.

I looked around re: error messages and did my best re: what I report and how I style it.

### CODE REVIEW

File: R/build-readme.R [hunk 1]
```diff
@@ -19,18 +19,26 @@ build_rmd <- function(files, path = ".", output_options = list(), ..., quiet = T
...
 
+  paths <- files
+  abs_files <- is_absolute_path(files)
+  paths[!abs_files] <- path(pkg$path, files[!abs_files])
+
+  ok <- file_exists(paths)
+  if (any(!ok)) {
+    stop(
+      "Can't find file(s): ",
+      paste0(encodeString(files[!ok], quote = "'"), collapse = ", "), ".",
```
Comment (jimhester) [inline R/build-readme.R:30]: This should use cli and/or rlang::abort I think.

Comment (jennybc) [reply R/build-readme.R:30]: Are you happy to bump minimum version of cli (now: 2.4.0) up to 3.0.0 to get `cli::cli_abort()`?

Comment (jennybc) [reply R/build-readme.R:30]: Ah, I guess this tells me the answer is YES: https://github.com/r-lib/devtools/issues/2364.

**Review by jimhester — APPROVED** [review]

File: R/build-readme.R (hunk 1 above)
Comment (jimhester) [reply R/build-readme.R]: That cleaned this up nicely!

**Pull request merged**